homeassistant/components/sensor/gearbest.py

```diff
@@ -0,0 +1,137 @@
+"""
+Parse prices of a item from gearbest.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.gearbest/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['gearbest_parser==1.0.5']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ITEMS = 'items'
+CONF_CURRENCY = 'currency'
+
+ICON = 'mdi:coin'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=2*60*60)  # 2h
+MIN_TIME_BETWEEN_CURRENCY_UPDATES = timedelta(seconds=12*60*60)  # 12h
+
+DOMAIN = 'gearbest'
+
+_ITEM_SCHEMA = vol.All(
+    vol.Any(
```

Comment:
Just a nitpick: This can be replaced by `has_at_least_one_key` from `config_validation.py`.

Reply:
Sure? It should validate if either url or id is present. It will also work when both are available, but it would make no sense?

Reply:
Oh no, you would still need the `vol.Exclusive()` below. Just the first `vol.Any(…)` can be replaced. You could of course build a `has_exactly_n_keys` in `config_validation.py` (which would look very similar to `has_at_least_one_key`) and use it then here.

Reply:
allright - you mean like
```
_ITEM_SCHEMA = vol.All(
    vol.Schema({
        vol.Exclusive('url', 'XOR'): cv.string,
        vol.Exclusive('id', 'XOR'): cv.string,
        vol.Optional("name"): cv.string,
        vol.Optional("currency"): cv.string
    }), cv.has_at_least_one_key("url", "id")
)
```
I'll test that later and commit it